File: android/app/src/main/java/com/example/data/Entities.kt
package com.example.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "patients")
data class Patient(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val age: Int,
    val gender: String,
    val email: String,
    val phone: String,
    val riskStatus: String, // "None", "Low", "Moderate", "Severe"
    val specialty: String, // e.g., "Depression", "Anxiety", "PTSD", "ADHD"
    val registrationDate: Long = System.currentTimeMillis()
)

@Entity(tableName = "appointments")
data class Appointment(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val patientId: Long,
    val patientName: String,
    val dateTime: String, // e.g., "Today, 10:00 AM" or "08 Jun 2026, 02:00 PM"
    val status: String, // "Scheduled", "Completed", "Cancelled"
    val notes: String = "",
    val fee: Double = 150.0,
    val isVideo: Boolean = true,
    val code: String = "", // teletherapy code e.g. "PSY-PYR-992"
    val practitionerName: String = "Dr. Katherine Brewster"
)

@Entity(tableName = "clinical_notes")
data class ClinicalNote(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val patientId: Long,
    val title: String, // e.g., "SOAP Note", "MSE Assessment", "Treatment Plan"
    val noteType: String, // "SOAP", "MSE", "PLAN", "GENERAL"
    val bodyJson: String, // JSON payload representing structured fields
    val timestamp: Long = System.currentTimeMillis(),
    val isRiskAlert: Boolean = false,
    val riskDisclaimer: String = "AI-assisted note. Licenses practitioner has reviewed."
)

@Entity(tableName = "assessments")
data class AssessmentScore(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val patientId: Long,
    val type: String, // "PHQ-9", "GAD-7", "DASS-21"
    val score: Int,
    val details: String, // e.g. "Mild Depression", "Severe Anxiety"
    val date: Long = System.currentTimeMillis()
)

@Entity(tableName = "mood_logs")
data class MoodLog(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val patientId: Long = 1, // Default logged-in patient
    val moodScore: Int, // 1 to 10
    val moodNote: String,
    val gratitude: String = "",
    val breathingSeconds: Int = 0,
    val date: Long = System.currentTimeMillis()
)

@Entity(tableName = "security_audit_logs")
data class SecurityAuditLog(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val action: String, // e.g., "Read Patient EHR", "Generated SOAP Note via AI"
    val details: String, // HIPAA Event description details
    val timestamp: Long = System.currentTimeMillis(),
    val actor: String = "Dr. Katherine Brewster (Admin)",
    val ipAddress: String = "192.168.1.104",
    val encryptionStandard: String = "AES-GCM-256"
)

@Entity(tableName = "homework_tasks")
data class Homework(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val patientId: Long,
    val patientName: String,
    val title: String,
    val description: String,
    val dueDate: String, // e.g. "15 Jun, 2026"
    val status: String, // "Assigned", "In Progress", "Completed"
    val patientNotes: String = "",
    val professionalFeedback: String = "",
    val assignedDate: Long = System.currentTimeMillis()
)

@Entity(tableName = "medications")
data class Medication(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val patientId: Long,
    val patientName: String,
    val name: String,
    val dosage: String,         // e.g. "50mg"
    val frequency: String,      // e.g. "Once daily (Morning)"
    val purpose: String,        // e.g. "Anxiety management"
    val instructions: String,   // e.g. "Take with food"
    val startDate: String,      // e.g. "09 Jun, 2026"
    val durationDays: Int = 30, // Prescription duration
    val isActive: Boolean = true,
    val prescribedBy: String = "Dr. Katherine Brewster"
)

@Entity(tableName = "adherence_logs")
data class AdherenceLog(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val medicationId: Long,
    val patientId: Long,
    val dateString: String,      // e.g. "09 Jun, 2026"
    val status: String,          // "Taken", "Missed"
    val timestamp: Long = System.currentTimeMillis()
)

@Entity(tableName = "scratchpad_notes")
data class ScratchpadNote(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val text: String,
    val timestamp: Long = System.currentTimeMillis(),
    val tag: String = "General"
)

@Entity(tableName = "user_accounts")
data class UserAccount(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val username: String,
    val email: String,
    val passwordPlain: String,
    val role: String, // "Professional" or "Patient"
    val fullName: String,
    val licenseOrId: String = "",
    val specialty: String = "",
    val bio: String = "No bio provided yet. Update your professional profile builder to tell patients more about your background and clinical methods.",
    val experienceYears: Int = 5,
    val clinicAddress: String = "Suite 300, Integrative Psychiatric Group, NY",
    val consultationFee: Double = 150.0,
    val rating: Double = 4.8,
    val availableHours: String = "9:00 AM - 5:00 PM",
    val languagesSpoken: String = "English"
)

@Entity(tableName = "availability_slots")
data class AvailabilitySlot(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val practitionerName: String,
    val dateString: String,      // format: "YYYY-MM-DD" e.g. "2026-06-15"
    val timeSlot: String,        // e.g. "09:00 AM", "10:30 AM", "01:00 PM"
    val isBooked: Boolean = false,
    val patientId: Long? = null,
    val patientName: String? = null,
    val notes: String = "",
    val isVideo: Boolean = true,
    val fee: Double = 150.0
)

@Entity(tableName = "peer_groups")
data class PeerGroup(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val description: String,
    val category: String, // e.g., "Anxiety", "Depression", "PTSD", "ADHD Coping", "General Wellness"
    val facilitatorName: String = "",
    val createdAt: Long = System.currentTimeMillis()
)

@Entity(tableName = "peer_messages")
data class PeerMessage(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val groupId: Long,
    val senderName: String,
    val senderRole: String,
    val messageText: String,
    val isAnonymous: Boolean = false,
    val isPinned: Boolean = false,
    val likesCount: Int = 0,
    val timestamp: Long = System.currentTimeMillis()
)

File: ios/PsyPyrus/Models/Entities.swift
import Foundation

public struct Patient: Identifiable, Codable {
    public var id: Int64
    public var name: String
    public var age: Int
    public var gender: String
    public var email: String
    public var phone: String
    public var riskStatus: String // "None", "Low", "Moderate", "Severe"
    public var specialty: String // e.g., "Depression", "Anxiety", "PTSD", "ADHD"
    public var registrationDate: Date
    
    public init(id: Int64 = 0, name: String, age: Int, gender: String, email: String, phone: String, riskStatus: String, specialty: String, registrationDate: Date = Date()) {
        self.id = id
        self.name = name
        self.age = age
        self.gender = gender
        self.email = email
        self.phone = phone
        self.riskStatus = riskStatus
        self.specialty = specialty
        self.registrationDate = registrationDate
    }
}

public struct Appointment: Identifiable, Codable {
    public var id: Int64
    public var patientId: Int64
    public var patientName: String
    public var dateTime: String // e.g., "Today, 10:00 AM"
    public var status: String // "Scheduled", "Completed", "Cancelled"
    public var notes: String
    public var fee: Double
    public var isVideo: BooleanLiteralType
    public var code: String // e.g. "PSY-PYR-992"
    
    public init(id: Int64 = 0, patientId: Int64, patientName: String, dateTime: String, status: String = "Scheduled", notes: String = "", fee: Double = 150.0, isVideo: Bool = true, code: String = "") {
        self.id = id
        self.patientId = patientId
        self.patientName = patientName
        self.dateTime = dateTime
        self.status = status
        self.notes = notes
        self.fee = fee
        self.isVideo = isVideo
        self.code = code
    }
}

public struct ClinicalNote: Identifiable, Codable {
    public var id: Int64
    public var patientId: Int64
    public var title: String
    public var noteType: String // "SOAP", "MSE", "PLAN", "GENERAL"
    public var bodyJson: String // Rich content or markdown text
    public var timestamp: Date
    public var isRiskAlert: Bool
    public var riskDisclaimer: String
    
    public init(id: Int64 = 0, patientId: Int64, title: String, noteType: String, bodyJson: String, timestamp: Date = Date(), isRiskAlert: Bool = false, riskDisclaimer: String = "AI-assisted note. Licensed practitioner has reviewed.") {
        self.id = id
        self.patientId = patientId
        self.title = title
        self.noteType = noteType
        self.bodyJson = bodyJson
        self.timestamp = timestamp
        self.isRiskAlert = isRiskAlert
        self.riskDisclaimer = riskDisclaimer
    }
}

public struct AssessmentScore: Identifiable, Codable {
    public var id: Int64
    public var patientId: Int64
    public var type: String // "PHQ-9", "GAD-7", "DASS-21"
    public var score: Int
    public var details: String // e.g. "Mild Depression"
    public var date: Date
    
    public init(id: Int64 = 0, patientId: Int64, type: String, score: Int, details: String, date: Date = Date()) {
        self.id = id
        self.patientId = patientId
        self.type = type
        self.score = score
        self.details = details
        self.date = date
    }
}

public struct MoodLog: Identifiable, Codable {
    public var id: Int64
    public var patientId: Int64
    public var moodScore: Int // 1 to 10
    public var moodNote: String
    public var gratitude: String
    public var breathingSeconds: Int
    public var date: Date
    
    public init(id: Int64 = 0, patientId: Int64 = 1, moodScore: Int, moodNote: String, gratitude: String = "", breathingSeconds: Int = 0, date: Date = Date()) {
        self.id = id
        self.patientId = patientId
        self.moodScore = moodScore
        self.moodNote = moodNote
        self.gratitude = gratitude
        self.breathingSeconds = breathingSeconds
        self.date = date
    }
}

public struct SecurityAuditLog: Identifiable, Codable {
    public var id: Int64
    public var action: String
    public var details: String
    public var timestamp: Date
    public var actor: String
    public var ipAddress: String
    public var encryptionStandard: String
    
    public init(id: Int64 = 0, action: String, details: String, timestamp: Date = Date(), actor: String = "Dr. Katherine Brewster (Admin)", ipAddress: String = "192.168.1.104", encryptionStandard: String = "AES-GCM-256") {
        self.id = id
        self.action = action
        self.details = details
        self.timestamp = timestamp
        self.actor = actor
        self.ipAddress = ipAddress
        self.encryptionStandard = encryptionStandard
    }
}

public struct HomeworkTask: Identifiable, Codable {
    public var id: Int64
    public var patientId: Int64
    public var description: String
    public var isCompleted: Bool
    public var assignedDate: Date
    
    public init(id: Int64 = 0, patientId: Int64, description: String, isCompleted: Bool = false, assignedDate: Date = Date()) {
        self.id = id
        self.patientId = patientId
        self.description = description
        self.isCompleted = isCompleted
        self.assignedDate = assignedDate
    }
}

public struct AppPlugin: Identifiable, Codable {
    public var id: String
    public var title: String
    public var category: String
    public var description: String
    public var price: String
    public var icon: String
    public var rating: Double
    public var installs: String
    public var isInstalled: Bool
    
    public init(id: String, title: String, category: String, description: String, price: String, icon: String, rating: Double, installs: String, isInstalled: Bool = false) {
        self.id = id
        self.title = title
        self.category = category
        self.description = description
        self.price = price
        self.icon = icon
        self.rating = rating
        self.installs = installs
        self.isInstalled = isInstalled
    }
}

public struct IcdSearchResult: Identifiable, Codable {
    public var id = UUID()
    public var code: String
    public var title: String
    public var uri: String
    
    public init(code: String, title: String, uri: String) {
        self.code = code
        self.title = title
        self.uri = uri
    }
}
